### DISCUSSION TITLE
Syntax highlighting doesn't work in diff kitten

### DISCUSSION THREAD

**Discussion by ROFLailXGOD** [kovidgoyal/kitty#4112] [General]
I've recently installed kitty and decided to try the diff kitten. The documentation says that `for syntax highlighting to work, pygments must be installed`, so I did that. I use pyenv for managing python installations, so it is located here:

```bash
➜  ~ which python
/home/vofel/.pyenv/shims/python
```

Of course, it is in the PATH. As for the version, I use the latest stable one which is 3.10.0. Pygments is installed:

```bash
➜  ~ pip list | grep Pygments
Pygments   2.10.0
```

So with this configuration I expect syntax highlighting to work, but it doesn't. Tried on 2 `.py` files. Using `pygmentize file.py`  directly from the command line works. So I'm not sure if the problem is on my side or if it is an actual bug, so I figured I'll ask here first.
Here's the debug output:

```
kitty 0.23.1 (cd7b4fcd8e) created by Kovid Goyal
Linux VofelPC 5.11.0-37-generic #41~20.04.2-Ubuntu SMP Fri Sep 24 09:06:38 UTC 2021 x86_64
Ubuntu 20.04.3 LTS \n \l
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"
Running under:X11
Loaded config files:
  /home/vofel/.config/kitty/kitty.conf

Config options different from defaults:
enable_audio_bell   False
font_family         Fira Code Medium
font_features:
{'FiraCode-Medium': ('+zero', '+onum')}
tab_activity_symbol ►
tab_bar_edge        1
tab_bar_style       powerline
tab_fade            (0.2, 0.4, 0.6, 0.8, 1.0)
tab_powerline_style slanted
Added shortcuts:
    shift+super+w KeyAction('close_os_window')
```

**Comment by kovidgoyal** [comment]
Putting it in PATH does not work. It needs to be in PYTHONPATH. I
suggest you use the official kitty binary builds, they come with
pygmentize bundled. https://sw.kovidgoyal.net/kitty/binary/ 

On Mon, Oct 11, 2021 at 04:42:11AM -0700, Vofel wrote:
> I've recently installed kitty and decided to try the diff kitten. The documentation says that `for syntax highlighting to work, pygments must be installed`, so I did that. I use pyenv for managing python installations, so it is located here:
> 
> ```bash
> ➜  ~ which python
> /home/vofel/.pyenv/shims/python
> ```
> 
> Of course, it is in the PATH. As for the version, I use the latest stable one which is 3.10.0. Pygments is installed:
> 
> ```bash
> ➜  ~ pip list | grep Pygments
> Pygments   2.10.0
> ```
> 
> So with this configuration I expect syntax highlighting to work, but it doesn't. Tried on 2 `.py` files. Using `pygmentize file.py`  directly from the command line works. So I'm not sure if the problem is on my side or if it is an actual bug, so I figured I'll ask here first.
> Here's the debug output:
> 
> ```
> kitty 0.23.1 (cd7b4fcd8e) created by Kovid Goyal
> Linux VofelPC 5.11.0-37-generic #41~20.04.2-Ubuntu SMP Fri Sep 24 09:06:38 UTC 2021 x86_64
> Ubuntu 20.04.3 LTS \n \l
> DISTRIB_ID=Ubuntu
> DISTRIB_RELEASE=20.04
> DISTRIB_CODENAME=focal
> DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"
> Running under:X11
> Loaded config files:
>   /home/vofel/.config/kitty/kitty.conf
> 
> Config options different from defaults:
> enable_audio_bell   False
> font_family         Fira Code Medium
> font_features:
> {'FiraCode-Medium': ('+zero', '+onum')}
> tab_activity_symbol ►
> tab_bar_edge        1
> tab_bar_style       powerline
> tab_fade            (0.2, 0.4, 0.6, 0.8, 1.0)
> tab_powerline_style slanted
> Added shortcuts:
>     shift+super+w KeyAction('close_os_window')
> ```
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/4112

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by ROFLailXGOD** [reply]
I actually installed kitty that way (Binary install section). I was mislead by this line `note that pygments is included in the macOS kitty app` on the kitty-diff page, because to me it sounds like I had to install pygments manually on Linux. Either way, I still can't manage to make it work. What exactly do I need to set PYTHONPATH to?

**Comment by kovidgoyal** [comment]
On Mon, Oct 11, 2021 at 06:05:37AM -0700, Vofel wrote:
> I actually installed kitty that way (Binary install section). I was mislead by this line `note that pygments is included in the macOS kitty app` on the kitty-diff page, because to me it sounds like I had to install pygments manually on Linux. Either way, I still can't manage to make it work. What exactly do I need to set PYTHONPATH to?

If you are using the binary you dont need to do anything it willl work
out of the box. Just make sure you have only a single kitty executable
on your system that points to the installed official binary.

You can test it with

kitty +runpy 'import pygments; print(pygments)'

**Reply by ROFLailXGOD** [reply]
That throws an error. Importing things like os works.

```
➜  ~ kitty +runpy 'import pygments; print(pygments)'
Traceback (most recent call last):
  File "runpy.py", line 197, in _run_module_as_main
  File "runpy.py", line 87, in _run_code
  File "kitty_main.py", line 153, in <module>
  File "kitty_main.py", line 144, in main
  File "kitty_main.py", line 99, in namespaced
  File "kitty_main.py", line 30, in runpy
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'pygments'
➜  ~ kitty +runpy 'import os; print(os)'      
<module 'os' (/home/vofel/.local/kitty.app/lib/kitty-extensions/python-lib.bypy.frozen/os.pyc)>
```

I had another kitty binary from the apt-get, but the version was old so I removed it and moved to the binary build.

```
➜  ~ sudo find / -name "kitty"
[sudo] password for vofel:
/home/vofel/.config/kitty
/home/vofel/.local/kitty.app/lib/kitty
/home/vofel/.local/kitty.app/lib/python3.9/kitty
/home/vofel/.local/kitty.app/bin/kitty
/home/vofel/.local/kitty.app/share/doc/kitty
/home/vofel/.local/bin/kitty
/home/vofel/.cache/kitty
➜  ~ which kitty
/home/vofel/.local/bin/kitty
➜  ~ ls -l /home/vofel/.local/bin/kitty                       
lrwxrwxrwx 1 vofel vofel 38 oct 6 19:53 /home/vofel/.local/bin/kitty -> /home/vofel/.local/kitty.app/bin/kitty
```

**Comment by kovidgoyal** [comment]
Looks like a bug, it will be fixed in the next kitty binary release.

**Reply by ROFLailXGOD** [reply]
So after all it's not me going crazy. Thanks for your time

**Reply by kovidgoyal** [reply]
https://github.com/kovidgoyal/kitty/commit/563b7ad2d0b2ca15beef15bf364b9bff8f27112d
